src/Parser/Hunter/Shared/Items/SoulOfTheHuntmaster.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+
+import SPECS from 'common/SPECS';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ItemLink from "common/ItemLink";
+
+import SUGGESTION_IMPORTANCE from 'Parser/Core/ISSUE_IMPORTANCE';
+
+const debug = false;
+
+class SoulOfTheHuntmaster extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFinger(ITEMS.SOUL_OF_THE_HUNTMASTER.id);
+    //Checks which spec has the ring equipped and then sets option1 or option2 accordingly - aswell as sets up the check for if they've picked another talent
+    switch (this.combatants.selected.spec) {
+      case SPECS.MARKSMANSHIP_HUNTER:
+        debug && console.log('SPEC DETECTED AS MM');
+        this.talentGained = SPELLS.LOCK_AND_LOAD_TALENT.id;
+        this.option1 = SPELLS.TRUE_AIM_TALENT.id;
+        this.option2 = SPELLS.BLACK_ARROW_TALENT.id;
+        this.hasPickedOtherTalent = this.combatants.selected.hasTalent(this.option1) || this.combatants.selected.hasTalent(this.option2);
```

Comment:
The check for other talent could be moved outside of the switch, to reduce the repeating.